### PR TITLE
HIVE-24575: VectorGroupByOperator reusing keys can lead to wrong results

### PR DESCRIPTION
**Pull request by dengzhhu653**
### Why are the changes needed?
VectorGroupByOperator reusing keys can lead to wrong results

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Local machine

### CODE REVIEW

File: ql/src/java/org/apache/hadoop/hive/ql/exec/vector/wrapper/VectorHashKeyWrapperGeneral.java [hunk 1]
```diff
@@ -348,8 +348,8 @@ public void copyKey(KeyWrapper oldWrapper) {
         }
       } else {
         System.arraycopy(byteLengths, 0, clone.byteLengths, 0, byteValues.length);
-        Arrays.fill(byteStarts, 0);
-        System.arraycopy(byteStarts, 0, clone.byteStarts, 0, byteValues.length);
+        Arrays.fill(clone.byteStarts, 0);
+        // System.arraycopy(byteStarts, 0, clone.byteStarts, 0, byteValues.length);
```
Comment (mustafaiman) [inline ql/src/java/org/apache/hadoop/hive/ql/exec/vector/wrapper/VectorHashKeyWrapperGeneral.java:352]: Please remove the line instead of commenting out.

**Review by mustafaiman — COMMENTED** [review]
Can you add a test that fails without the patch so we prevent further regressions in the future?

File: ql/src/java/org/apache/hadoop/hive/ql/exec/vector/wrapper/VectorHashKeyWrapperGeneral.java (hunk 1 above)
Comment (dengzhhu653) [reply ql/src/java/org/apache/hadoop/hive/ql/exec/vector/wrapper/VectorHashKeyWrapperGeneral.java]: Fixed and added a test for this. Thanks very much for the review!

**Comment by mustafaiman** [comment]
+1 pending tests

**Comment by dengzhhu653** [comment]
Thanks much for the help and review, @mustafaiman.  close it

**Pull request closed without merging**